packages/keystone/bin/utils.js

```diff
@@ -59,44 +59,71 @@ function extractAppMeta(apps) {
 async function executeDefaultServer(args, entryFile, distDir, spinner) {
   const port = args['--port'] ? args['--port'] : DEFAULT_PORT;
   const connectTo = args['--connect-to'] ? args['--connect-to'] : DEFAULT_CONNECT_TO;
+  let router;
+  let status = 'start-server';
 
+  spinner.text = 'Starting Keystone server';
+  const app = express();
+
+  app.use((req, res, next) => {
+    if (router) {
+      return router(req, res, next);
+    } else {
+      res.format({
+        default: () => res.sendFile(path.resolve(__dirname, './loading.html')),
+        'text/html': () => res.sendFile(path.resolve(__dirname, './loading.html')),
+        'application/json': () => res.json({ loading: true, status }),
+      });
+    }
+  });
+
+  const { server } = await new Promise((resolve, reject) => {
+    const server = app.listen(port, error => {
+      if (error) {
+        return reject(error);
+      }
+      return resolve({ server });
+    });
+  });
+
+  spinner.succeed(`Keystone server listening on port ${port}`);
   spinner.text = 'Initialising Keystone instance';
 
+  status = 'init-keystone';
+
   // Allow the spinner time to flush its output to the console.
   await new Promise(resolve => setTimeout(resolve, 100));
 
   const { keystone, apps = [] } = require(path.resolve(entryFile));
 
   spinner.succeed('Initialised Keystone instance');
-  spinner.start('Preparing Keystone server');
+  spinner.start('Connecting to database');
+
+  status = 'db-connect';
 
-  const app = express();
   const dev = process.env.NODE_ENV !== 'production';
 
   const { middlewares } = await keystone.prepare({ apps, port, distDir, dev });
```

Comment:
This could be re-written so that we connect to the database (first) but also start preparing apps in parallel.

Remembering that we now have an `onConnect` method, which can do things like create initial data from imports, it would be good to kick this off as early as possible.

By waiting for both promises to resolve asynchronously, we would then likely (in most cases) connect first (this is pretty instant) then go into "preparing to accept requests" mode while the apps' `prepare` promises are being resolved - which is _slightly_ more correct than what's displayed in the current setup, which will make "connecting to database" look like it's taking forever when what's actually happening is we're waiting on a build process. At least in dev.

This is mainly important because people might start wondering why connecting to the database is so slow, and try to fix it 🤷‍♂